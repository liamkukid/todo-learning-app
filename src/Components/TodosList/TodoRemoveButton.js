import PropTypes from 'prop-types';
import { useDispatch } from 'react-redux';
import { useRef } from 'react';
import style from './list.module.scss';
import gabarge from '../../icons/gabarge.svg';
import { todoRemove } from '../../features/todosSlice';

export default function TodoRemoveButton({ todoId }) {
  const dispatch = useDispatch();
  const buttonRef = useRef();
  const handleRemoveClick = () => {
    dispatch(todoRemove(buttonRef.current.value));
  };

  return (
    <button
      ref={buttonRef}
      data-testid="todo-remove"
      value={todoId}
      type="button"
      className={style.remove_todo_button}
      onClick={handleRemoveClick}
    >
      <img src={gabarge} className="gabarge-logo" alt="gabarge" />
    </button>
  );
}

TodoRemoveButton.propTypes = {
  todoId: PropTypes.string.isRequired,
};
