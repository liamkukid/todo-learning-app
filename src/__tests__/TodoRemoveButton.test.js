import renderer from 'react-test-renderer';
import { render, screen, fireEvent } from '@testing-library/react';
import { configureStore } from '@reduxjs/toolkit';
import { Provider } from 'react-redux';
import todosReducer from '../features/todosSlice';
import TodoRemoveButton from '../Components/TodosList/TodoRemoveButton';

describe('TodoRemoveButton', () => {
  let store;
  const todo = {
    title: 'todo 1',
    done: false,
    id: '14e520cd-63f2-485e-9b3b-072ce068554e',
  };

  beforeEach(() => {
    store = configureStore({
      reducer: {
        todos: todosReducer,
      },
      preloadedState: {
        todos: [todo],
      },
    });
  });

  afterEach(() => {
    store = null;
  });

  it('renders correctly', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <TodoRemoveButton todoId={todo.id} />
      </Provider>
    );
    expect(tree).toMatchSnapshot();
  });

  it('removes the todo', () => {
    render(
      <Provider store={store}>
        <TodoRemoveButton todoId={todo.id} />
      </Provider>
    );
    const linkElementButton = screen.getByTestId('todo-remove');

    fireEvent.click(linkElementButton);

    const state = store.getState();
    expect(state.todos).toEqual([]);
  });
});
