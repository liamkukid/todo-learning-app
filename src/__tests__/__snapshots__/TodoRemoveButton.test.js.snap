// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoRemoveButton renders correctly 1`] = `
<button
  className="remove_todo_button"
  data-testid="todo-remove"
  onClick={[Function]}
  type="button"
  value="14e520cd-63f2-485e-9b3b-072ce068554e"
>
  <img
    alt="gabarge"
    className="gabarge-logo"
    src="gabarge.svg"
  />
</button>
`;
